// quicktype-core/dist/language/Haskell/language.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HaskellTargetLanguage = exports.haskellOptions = void 0;
const RendererOptions_1 = require("../../RendererOptions");
const TargetLanguage_1 = require("../../TargetLanguage");
const HaskellRenderer_1 = require("./HaskellRenderer");
exports.haskellOptions = {
    justTypes: new RendererOptions_1.BooleanOption("just-types", "Plain types only", false),
    useList: new RendererOptions_1.EnumOption("array-type", "Use Array or List", [
        ["array", false],
        ["list", true]
    ]),
    moduleName: new RendererOptions_1.StringOption("module", "Generated module name", "NAME", "QuickType")
};
class HaskellTargetLanguage extends TargetLanguage_1.TargetLanguage {
    constructor() {
        super("Haskell", ["haskell"], "haskell");
    }
    getOptions() {
        return [exports.haskellOptions.justTypes, exports.haskellOptions.moduleName, exports.haskellOptions.useList];
    }
    get supportsOptionalClassProperties() {
        return true;
    }
    get supportsUnionsWithBothNumberTypes() {
        return true;
    }
    makeRenderer(renderContext, untypedOptionValues) {
        return new HaskellRenderer_1.HaskellRenderer(this, renderContext, (0, RendererOptions_1.getOptionValues)(exports.haskellOptions, untypedOptionValues));
    }
}
exports.HaskellTargetLanguage = HaskellTargetLanguage;
